src/coreclr/jit/codegenxarch.cpp

```diff
@@ -10307,6 +10315,69 @@ void CodeGen::genPushCalleeSavedRegisters()
     }
 #endif // DEBUG
 
+#ifdef TARGET_AMD64
+    if (compiler->canUseApxEncoding() && compiler->canUseEvexEncoding() && JitConfig.EnableApxPPX())
+    {
+        // This is not a funclet or an On-Stack Replacement.
+        assert((compiler->funCurrentFunc()->funKind == FuncKind::FUNC_ROOT) && !compiler->opts.IsOSR());
+        // PUSH2 doesn't work for ESP.
+        assert((rsPushRegs & RBM_SPBASE) == 0);
```

Comment:
any reason why this code is not in its own method `genPushCalleeSavedRegistersFromMaskAPX` similar to `genPopCalleeSavedRegistersFromMaskAPX`?